tests/components/ShareForm/ShareForm.test.jsx

```diff
@@ -0,0 +1,77 @@
+import { render, screen } from '@testing-library/react';
+import ShareForm from '../../../src/components/ShareForm';
+import * as FirebaseFunctions from '../../../src/api/firebase';
+import { vi } from 'vitest';
+import { QueryClientProvider } from 'react-query';
+import { queryClient } from '../../../src/App';
+import userEvent from '@testing-library/user-event';
+
+const sleep = async (ms = 200) =>
+	await new Promise((resolve) => setTimeout(() => resolve(null), ms));
+
+describe('ShareForm', () => {
+	afterEach(() => {
+		vi.resetAllMocks();
+		vi.clearAllMocks();
+	});
+
+	const renderShareListForm = ({ listPath }) => {
+		return render(
+			<QueryClientProvider client={queryClient}>
+				<ShareForm listPath={listPath} />
+			</QueryClientProvider>,
+		);
+	};
+
+	it('renders ShareForm', () => {
+		renderShareListForm({ listPath: '/test-list' });
+	});
+
+	// it('Shares form with user successfully', async () => {
```

Comment:
nit: remove comment if not addressing this 😄

Reply:
hi @bdanzer , i actually wanted to ask you about this so had it commented out. when I try to run this test I am getting the error `Cannot read properties of undefined (reading 'uid')`, but can't seem to figure out how to a
<img width="1054" alt="Screenshot 2024-03-10 at 6 47 27 PM" src="https://github.com/the-collab-lab/tcl-67-smart-shopping-list/assets/46732901/1bf01e44-0af8-4914-9110-13be9eb2ddd5">
ddress it.

Reply:
@jtkabenni Looking at ShareForm it looks like you'll also need to spyOn the useAuth hook and mock the user for this one. Also in your test, you'll want to confirm what `mockedShareList` args was called with. Just as a heads up this is the [vitest docs site for mocks](https://vitest.dev/api/mock.html) and methods available for expect [docs page](https://vitest.dev/api/expect.html). It's based pretty much on what jest does. They have a lot of good examples to refer to as well.

Reply:
thanks @bdanzer , is the idea that I run the `mockedShareList` with the `mockedUser` as a parameter? How do I ensure that the `mockedShareList` is called with`mockedUser`?
```
		const mockedShareListReturnValue = 'List was shared successfully';
		const mockedShareList = vi.spyOn(FirebaseFunctions, 'shareList');
		mockedShareList.mockImplementationOnce(async () => {
			await mockedUser();
			await sleep();
			return mockedShareListReturnValue;
		});
```

Reply:
@jtkabenni  inside of ShareForm there is this import:

```
import { useAuth } from '../api';
```

so in your test you'll want to do another spy on that. Basically the spys are to override the exports of those files and to attach your own mock functions on to them. So you'll want to change the return value of `useAuth`.

But you wouldn't be called mockedUser inside of `mockedShareList`

Reply:
The way shareList was mocked is very close to what you need to do for useUser

Reply:
```		
const mockedUserReturnValue = {
			uid: 'testuser',
			email: 'test@gmail.com',
		};
		const mockedUser = vi.spyOn(AuthFunctions, 'useAuth');
		mockedUser.mockImplementationOnce(async () => {
			await sleep();
			return mockedUserReturnValue;
		});

		// mocked share list
		const mockedShareListReturnValue = 'List was shared successfully';
		const mockedShareList = vi.spyOn(FirebaseFunctions, 'shareList');
		mockedShareList.mockImplementationOnce(async (<mockeduser>??) => {
			await sleep();
			return mockedShareListReturnValue;
		});
		
```
I'm just not sure on how to include the `mockedUser` in my `mockedShareList` in a way where it will be used when called 

@bdanzer

Reply:
@jtkabenni In this case you can ignore the params in the function. The way to check what values it was called with would be this:

```javascript
expect(mockedShareList).toHaveBeenCalledWith() //validate the arguments and the user id should be passed as an argument 
```

Let me know if that makes sense. Everything above looks good just keep the mockImplementation like this:

```
mockedShareList.mockImplementationOnce(async () => {
     await sleep();
     return mockedShareListReturnValue;
});
```

And add the expect that I mentioned

Reply:
Thanks @bdanzer , seems like with some furious debugging I figured out what was going on: 

Return value for user I had set as
```
		const mockedUserReturnValue = user: {
				uid: 'testuser',
			},

```
instead of 
```
		const mockedUserReturnValue = {
			user: {
				uid: 'testuser',
			},
		};
```


I also had 
```
	mockedUser.mockImplementation(async () => {
			await sleep();
			return mockedUserReturnValue;
		});
```

when since the mockedUserReturnValue wasn't a promise so I didn't need to wait for it. 
```
	mockedUser.mockImplementation( () => {

			return mockedUserReturnValue;
		});

```

I think most of all I was confused on when mockedUser was getting called, which I then figured out it was after the shareForm gets rendered.